server/crawler/index.ts

```diff
@@ -195,25 +195,79 @@ export default async function getCralwerData(url: string) {
   let title = ''
   let body = ''
   let image = ''
+  const imgDataKey:any = []
+
   try {
     const { data: content } = await axios.get(encodeURI(url as string))
+    const { protocol , host} = new URL(url)
     const $ = cheerio.load(content);
     [
       'script',
       'style',
       'svg',
+      'source',
     ].forEach((t: string) => $(t).remove())
     title = $('title').text()
     const metas = $('meta')
     body = $('body').html() || ''
+    const promiseImg:any = []
+    // const promiseSource:any = []
+    const imgKey:any = []
+    // const sourceKey:any = []
 
     const img = $('img')
     Object.keys(img).forEach((key: any) => {
       const { src } = img[key].attribs || {};
       if (!src) {
         delete img[key]
+      } else {
+        let srcUrl = src
+        if ( !src.startsWith('http') ) {
+          srcUrl = `${protocol}//${host}${src}`
+        }
+        promiseImg.push(axios.get(`${srcUrl}`, {responseType: 'arraybuffer'})
+        .then((element)=> {
+          const srcChange = src.replace(/&/g, `&amp;`)
```

Comment:
Would using `.map` + `Promise.all` get us a simpler structure than `.forEach` and `.then`?